### PR TITLE
NetworkManager nixos: add noDns option 

### PR DESCRIPTION
**Pull request by jD91mZM2**
No touchy tha DNS!
Would it be better to add a general `dns` enum?

### CODE REVIEW

**Comment by peterhoeg** [comment]
Being able to set conflicting options like ```noDns``` and ```useDnsmasq``` isn't great. One option could be to an enum as you suggest with a ```system``` option that applies the logic currently in place for ```systemd-resolved``` and ```unbound```. The alternative is to get rid of the automatic detection and turn the dependency upside down so that setting ```dns``` to ```systemd-resolved``` automatically enabled it.

**Comment by jD91mZM2** [comment]
Maybe the automatic detection could enabled only if `dns` is null?

**Comment by peterhoeg** [comment]
That doesn't seem intuitive. How about an ```auto``` option which is the default?

**Comment by jD91mZM2** [comment]
Even better! I'll get to work! Do I keep but deprecate the old useDnsmasq option?

**Comment by peterhoeg** [comment]
You can deprecate by adding it here: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/rename.nix and then removing it accordingly.

When deprecating an option please also add a notice to the release notes.

**Comment by jD91mZM2** [comment]
Aha, thanks!

File: nixos/modules/services/networking/networkmanager.nix [hunk 1]
```diff
@@ -205,14 +206,20 @@ in {
         };
       };
 
-      useDnsmasq = mkOption {
-        type = types.bool;
-        default = false;
+      dns = mkOption {
+        type = types.enum ["auto" "dnsmasq" "none"];
```
Comment (peterhoeg) [inline nixos/modules/services/networking/networkmanager.nix]: nitpick - can you add spaces inside the list ```[ "auto" "dnsmasq" "none" ]```

**Comment by peterhoeg** [comment]
And if you can fix the merge conflict, I'll go ahead and merge this.

**Comment by jD91mZM2** [comment]
@peterhoeg Awesome, thanks! Rebased :)

**Comment by peterhoeg** [comment]
Thank you for doing up the PR and the quick turnaround time!

**Pull request merged**